app/models/schema_version.rb

```diff
@@ -23,7 +23,27 @@ class SchemaVersion < ApplicationRecord
   scope :for_schema,
         ->(schema_id) { where(schema_id: schema_id) }
   scope :for_schema_fingerprint,
-        ->(fingerprint) { joins(:schema).where('schemas.fingerprint = ?', fingerprint) }
+        ->(fingerprint) do
```

Comment:
Moved into scopes in the `Schema` model. I don't think this can be used by `Schema.existing_schema` which wants to prefer results from `fingerprint2`.